fiftyone/core/collections.py

```
@@ -979,10 +1008,27 @@ def one(self, expr, exact=False):
-            a :class:`fiftyone.core.sample.Sample` or
-            :class:`fiftyone.core.sample.SampleView`
+            a :class:`fiftyone.core.sample.SampleView`
         """
-        raise NotImplementedError("Subclass must implement one()")
+        view = self.match(expr)
+        matches = iter(view)
+
+        try:
+            sample = next(matches)
+        except StopIteration:
+            raise ValueError("No samples match the given expression")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)